spock-core/src/main/java/org/spockframework/runtime/SpockNode.java

```diff
@@ -126,4 +130,34 @@ protected static ExecutionMode toExecutionMode(org.spockframework.runtime.model.
       .map(er -> new ExclusiveResource(er.getKey(), toLockMode(er.getMode())))
       .collect(Collectors.toSet());
   }
+
+  @Override
+  public Set<TestTag> getTags() {
+    // Lazy initialisation
+    if (tags == null) {
+      tags = new HashSet<>();
+      SpecInfo specInfo;
+      // Node info can be either FeatureInfo or SpecInfo
+      if (getNodeInfo() instanceof FeatureInfo) {
+        FeatureInfo featureInfo = (FeatureInfo) getNodeInfo();
+        addTestTagsFrom(featureInfo.getTags());
+        specInfo = featureInfo.getSpec();
+      }
```

Comment:
Actually, originally it was in subclasses, but then I decided to move it up, also in order to avoid having to either duplicate the private field or to make it protected, but leave it uninitialised in the base class. Both approaches have pros and cons, but in the end I also liked the fact that now there is a single place where you can just read a not very complicated method and understand what is happening. it is compact, and I wanted to keep the PR minimally invasive. The subclasses also do not need to care about the whole JUnit tag issue. I remembered your criticism to try and avoid to let JUnit bleed into Spock as much as possible, and this is why I implemented it this way. I also do not like `instanceof` very much, but it is not so unclean that I decided to break up and scatter the logic across subclasses.

Reply:
This point becomes moot, when moving the distribution/aggregation responsibility to the extension.

Reply:
Please note comment https://github.com/spockframework/spock/pull/1437#discussion_r823375850 for why that is a bad idea.

Reply:
Please note that 886dad34 moves aggregation to `ITaggable` and its implementing classes. The commit comment explains in detail why I did that and what the benefits are. For `SpockNode` it means more concise code and no more `instanceof` checks in this context.